File: src/Tests/Behavioral/ManualConversionSiblingState/state.cs
// Hand-owned manual conversion of state.go — the marker below drops state.go from the convert set,
// so a re-transpile must NOT overwrite this file (it emits the state.cs.auto review sibling instead)
// while STILL analyzing/visiting state.go so its package-wide state (the lifted schedlike_disable
// anonymous-struct type, the sched/newprocs package vars) reaches the sibling main.cs emission.
// This file guards the seeded-reconvert defect fixed 2026-07-17: skipping a marked file's entire
// visit corrupted every sibling file (raw Go struct{...} text in selectors, package-var assignments
// re-declared as shadowing locals).
[module: go.GoManualConversion]

namespace go;

partial class main_package {

[GoType("dyn")] partial struct schedlike_disable {
    internal bool user;
    internal int32 n;
}

[GoType] partial struct schedlike {
    internal schedlike_disable disable;
    internal @string label;
}

internal static ж<schedlike> Ꮡsched = new(default(schedlike));
internal static ref schedlike sched => ref Ꮡsched.Value;

internal static int32 newprocs;

} // end main_package

File: src/Tests/Behavioral/ManualConversionSiblingState/state.go
package main

// schedlike mirrors the runtime.schedt shape: a field whose type is an ANONYMOUS struct. The
// converter lifts the anonymous struct to a package-level C# type and registers the lifted name
// in the package-wide registry — state a marker-skipped visit must still contribute so sibling
// files can resolve cross-file references to it.
type schedlike struct {
	disable struct {
		user bool
		n    int32
	}
	label string
}

// Package-level state consumed by the SIBLING file (main.go): a global whose nested
// anonymous-struct field gets addressed there, and a package var assigned there.
var sched schedlike

var newprocs int32
